beacon-chain/core/state/transition.go

```diff
@@ -379,7 +377,7 @@ func CanProcessEpoch(state *pb.BeaconState) bool {
 //    process_slashings(state)
 //    process_final_updates(state)
 //    # @after_process_final_updates
-func ProcessEpoch(ctx context.Context, state *pb.BeaconState) (*pb.BeaconState, error) {
+func ProcessEpoch(ctx context.Context, state *pb.BeaconState, logging bool) (*pb.BeaconState, error) {
```

Comment:
Interesting pattern, how do we feel about passing logging flag to the stateless function? Is this a pattern we aim to move forward with?